pkg/process/monitor/process_monitor.go

```diff
@@ -249,11 +240,10 @@ func (pm *ProcessMonitor) Initialize() error {
 	go func() {
 		logTicker := time.NewTicker(2 * time.Minute)
 
-		terminated := false
 		defer func() {
-			if !terminated {
-				pm.terminateProcessMonitor()
-			}
+			log.Info("netlink process monitor ended")
+			pm.wg.Done()
+			pm.Stop()
```

Comment:
Looking at 2nd pass, it's not the good startegy here
we must cleanup explicitly here, (not depending on the refcount)

On API callpath all is fine
but if we are on the errors path (or done) (aka returns)
we should stop the callbacksRunners, cleanup chans, refcount